apps/rush-lib/src/data/VersionMismatchFinder.ts

```diff
@@ -14,12 +14,14 @@ export class VersionMismatchFinder {
   *   }
   * }
   */
+  private _allowedAlternativeVersion:  Map<string, ReadonlyArray<string>>;
   private _mismatches: Map<string, Map<string, string[]>>;
   private _projects: RushConfigurationProject[];
 
-  constructor(projects: RushConfigurationProject[]) {
+  constructor(projects: RushConfigurationProject[], allowedAlternativeVersions:  Map<string, ReadonlyArray<string>>) {
     this._projects = projects;
     this._mismatches = new Map<string, Map<string, string[]>>();
+    this._allowedAlternativeVersion = allowedAlternativeVersions;
```

Comment:
Is it optional? Should you throw if this isn't available?

Reply:
I don't think a copy needs to be made, this is the only consumer of the alternative versions. However I will give it a fallback of being an empty map if it isn't provided.